src/MongoDB.Analyzer/Core/ReferencesProvider.cs

```diff
@@ -172,6 +181,27 @@ private static (Version, IDictionary<string, string>, string MissingAssembly) Ge
         return (version, nameToPathMapping, null);
     }
 
+    private static void CompilePocoAnalysisCode(List<MetadataReference> metadataReferences)
```

Comment:
It's better to have `ReferencesProvider` not to be aware for specific analysis, so GetReferences needs to receive this poco reference as well. This code should be somewhere in Poco namespace.

Reply:
This code has been moved to MongoDB.Analyzer.Core.Poco namespace as suggested